### PR TITLE
(WIP) feat: query dynamodb for id mapping

### PR DESCRIPTION
**Pull request by ilhaoni**
Instead of using in memory mapping, we use AWS Athena to query instead. SQL Query is stored in seed.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
   targetEndpoint: process.env.TARGET_ENDPOINT || "",
   appParameters: {
-    salt: process.env.SALT || DEFAULT_SALT
+    salt: process.env.SALT || DEFAULT_SALT,
+    sqlQuery: process.env.SQL_QUERY || ""
   }
 });
 
+const athenaExpressConfig = { aws };
+export const athenaExpress = new AthenaExpress(athenaExpressConfig);
```
Comment (yehjxraymond) [inline src/config.ts:18]: Better to have this in another file `athena.ts` or something in the service folder (if there is one?)

File: src/config.ts [hunk 2]
```diff
@@ -1,11 +1,20 @@
+import aws from "aws-sdk";
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const AthenaExpress = require("athena-express");
```
Comment (yehjxraymond) [inline src/config.ts:4]: Why using require?

File: .env.example [hunk 3]
```diff
@@ -1,2 +1,3 @@
 TARGET_ENDPOINT=https://wfs7ytbybk.execute-api.ap-southeast-1.amazonaws.com/pr80
-SALT=salt
\ No newline at end of file
+SALT=salt
+SQL_QUERY=select * from something
```
Comment (yehjxraymond) [inline .env.example:3]: Hmm why did you choose to put SQL query in the env?

File: src/functionHandlers/proxy/idMapper.ts [hunk 4]
```diff
@@ -1,17 +1,7 @@
+/* eslint-disable no-console */
```
Comment (yehjxraymond) [inline src/functionHandlers/proxy/idMapper.ts:1]: rm

File: src/functionHandlers/proxy/idMapper.ts [hunk 5]
```diff
@@ -20,7 +10,17 @@ export const identifierStringToSaltedHash = (identifier: string) =>
...
+
+  try {
+    const queryResults = await athenaExpress.query(query);
+  
+    if (queryResults.Items.length === 0) throw new Error("Record does not exist");
+    const result = (queryResults.Items)[0].address;
+  
+    return result;
+  } catch (error) {
+    console.log(error);
```
Comment (yehjxraymond) [inline src/functionHandlers/proxy/idMapper.ts:23]: use the error boundary is needed, if not you will have a lot of stray console

File: src/config.ts (hunk 1 above)
Comment (ilhaoni) [reply src/config.ts:18]: dont have eh, i can create xD

File: src/config.ts (hunk 2 above)
Comment (ilhaoni) [reply src/config.ts:4]: i tried using import and it didnt work :x let me try again!

**Pull request closed without merging**